library/src/updater.rs

```diff
@@ -805,4 +834,45 @@ mod tests {
         })
         .unwrap();
     }
+
+    #[test]
+    fn no_config_lock_contention_when_waiting_for_patch_check() {
+        let tmp_dir = TempDir::new("example").unwrap();
+        init_for_testing(&tmp_dir, None);
+
+        // Set up the network hooks to sleep for 10 seconds on a patch check request
+        let hooks = NetworkHooks {
+            patch_check_request_fn: |_url, _request| {
+                let patch_check_delay = std::time::Duration::from_secs(2);
+                std::thread::sleep(patch_check_delay);
+                // If we get here, the test has failed.
+                unreachable!("If the test has not terminated before this, set_config is likely being blocked by a patch check request, which should not happen");
```

Comment:
Does the test terminating cause the check_for_update thread to get killed?  I'm not sure what code here is tearing down the thread before this gets hit?  A naive implementation I would assume this to just get hit *after* the test runs (while some other test is running).

Reply:
> Does the test terminating cause the check_for_update thread to get killed?

No, although I'm not sure what happens if the tests take longer than 2s to run. Let me dig into that a bit more.

Reply:
Added a bool to ensure that we don't hit `unreachable` if we're managed to obtain and release the lock